### DISCUSSION TITLE
Shopify App Proxy not working -> HTTPS error

### DISCUSSION THREAD

**Discussion by ayodyln** [edlaver/cloudflare-worker-partytown-shopify-app-proxy#2] [Q&A]
Hello @edlaver,

I'm a new Shopify developer and I'm trying to fix the speed scores on my companies Shopify website. The main culprit are the applications being used, but most of them are crucial to business operations. I'll remove any apps we don't need too.

But, I need to figure out how to use Partytown to improve the loading times on my Shopify Theme. I've read through the documentation and admittedly I'm new to Cloudflare, and now I'm stuck on the part were we set up the app proxy. I'm missing something here, because it keeps saying that my url I add to the Proxy configuration is not HTTPS. But the url contains the HTTPS prefix, is this an issue with my Cloudflare deployment?

What can I now?

**Comment by ayodyln** [comment]
## Update

The app's proxy URL all of the sudden was accepted as HTTPS. But now I'm curious about further clarification on how the integration process works.

I created an Shopify App **manually**, do I now install that on my shopify store?

**Reply by ayodyln** [reply]
Another follow-up question:

Why do we need to get rid of Google Analytics?

**Reply by edlaver** [reply]
Sounds a bit weird wrt the proxy, no idea why it wasn't accepting an HTTPS address, but sounds like that part is sorted now?

> I created an Shopify App manually, do I now install that on my shopify store?
Yeah, you can add it to a dev store from within the Partners Admin section in Shopify.

App > Overview > Test your app
![image](https://github.com/edlaver/cloudflare-worker-partytown-shopify-app-proxy/assets/56447343/e6fce14f-67e1-4252-9804-8a8fe8d3ca61)

Or as a custom app: https://help.shopify.com/en/manual/apps/app-types/custom-apps

> Why do we need to get rid of Google Analytics?
Need is perhaps too strong a word, but wrt Partytown - the idea is to improve the speed and responsiveness of your shop by moving non-essential JavaScript onto another thread (a "worker").

Google Analytics used to be one of the worst offenders for this.  But for analytics use-cases, you're probably now better off using Shopify's native "Pixels" functionality as opposed to Partytown: https://help.shopify.com/en/manual/promoting-marketing/pixels

Partytown could still be of use for non-analytics use cases where Pixels can't be used.
